packages/backend-auth/src/factory.ts

```diff
@@ -156,12 +161,18 @@ class AmplifyAuthGenerator implements ConstructContainerEntryGenerator {
     // this produces the access definition that will be used to create the storage policies
     const accessDefinition = this.props.access(this.allowAccessBuilder);
 
+    const ssmEnvironmentEntries =
+      ssmEnvironmentEntriesGenerator.generateSsmEnvironmentEntries({
+        [`${this.defaultName}_USERPOOL_ARN`]:
```

Comment:
Looks like the SDK uses the UserPool Id, not ARN. That's what we should set here

For example https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/cognito-identity-provider/command/AdminCreateUserCommand/